### DISCUSSION TITLE
Cannot set admin password

### DISCUSSION THREAD

**Discussion by hankhank10** [datalust/seq-tickets#1119] [Help]
I have followed the instructions to deploy to a linux docker but whatever I do when I access the server I cannot login.

<img width="536" alt="Screenshot 2021-02-03 at 20 11 45" src="https://user-images.githubusercontent.com/25515609/106803394-162cef00-665c-11eb-9630-6d4416d93f2e.png">

I have tried both setting an admin password using:

`PH=$(echo '<password>' | docker run --rm -i datalust/seq config hash)`

and then 

```
docker run \
  --name seq \
  -d \
  --restart unless-stopped \
  -e ACCEPT_EULA=Y \
  -e SEQ_FIRSTRUN_ADMINPASSWORDHASH="$PH" \
  -v /path/to/seq/data:/data \
  -p 80:80 \
  -p 5341:5341 \
  datalust/seq
```

and also not setting any authentication using:

```
docker run \
  --name seq \
  -d \
  --restart unless-stopped \
  -e ACCEPT_EULA=Y \
  -v /path/to/seq/data:/data \
  -p 80:80 \
  -p 5341:5341 \
  datalust/seq
```

I get the same error whenever I try to log in.

**Comment by nblumhardt** [comment]
Thanks for the heads-up! Those instructions were recently updated, so we might have missed something - we'll take a look!

In the meantime, if you don't have any important data in the instance yet, you can clear out the contents of the mounted volume and use the second command-line to start up without a password set.

**Reply by hankhank10** [reply]
Thank you.

I have tried clearing it out and rebuilding with the second command set but I am still prompted for a username and password. I am obviously doing something wrong!

I am running this on a VPS so connecting remotely - not sure if that might be having an impact?

Thanks

**Reply by nblumhardt** [reply]
It does sound like something else is going on, there - without the `SEQ_FIRSTRUN_ADMINPASSWORDHASH` variable, and with a clean data directory, you should get straight in.

Perhaps deleting the folder at `/path/to/seq/data` entirely will help?

**Comment by hankhank10** [comment]
I never fixed this.

In the end I managed to get in by way of using "auth" to set the admin password manually

**Comment by svarga91** [comment]
I have similar issue in docker on synology. After enabling auth I cannot login. Username and password are 100% correct. It is strange. 
 
![image](https://github.com/datalust/seq-tickets/assets/4550747/c916e87b-947a-4d51-b19a-67b84122b8c8)

instead of redirect to dashboard I am cycling in login paths

http://synology:5300/#/login?to=%2Flogin%3Fto%3D%252Flogin

**Reply by KodrAus** [reply]
Thanks for the report @svarga91. Are you able to share what version of Seq you're running?

**Reply by svarga91** [reply]
I tried Seq (2023.4.10146) on docker on windows and everything works fine. 
But when I tried it on Synology NAS (there is Container Manager from Docker Inc.) with Seq version (2023.4.10146) there is an issue after enable auth. 
I choose "Username and Password" and fill the rest fields. After submitting form I am redirected to login page and cannot login. I am able to reset password with init script but still I cannot login after recovery password. It looks like some issue on login page. In logs I see that I am logged in but I am again redirected to login.

See logs

[seq-20230925.log](https://github.com/datalust/seq-tickets/files/12712932/seq-20230925.log)

**Reply by svarga91** [reply]
When I setup docker on ubuntu desktop and install seq on "Docker Desktop" everything works. So there is an issue with seq on synology "Container Manager" docker.

**Reply by nblumhardt** [reply]
Thanks for the update, @svarga91.

Your best bet is completely removing the Seq container's storage on the NAS and starting from scratch. The `SEQ_FIRSTRUN_*` variables are only used on the very first execution of the container over a given `/data` mount, afterwards they'll be ignored, so any typos or other problems that interfered with the first execution will be permanent until the storage mount is cleared out.

We don't have any knowledge/expertise around running Seq on a NAS; it's cool to hear that you're giving it a shot, but you may need to tinker a bit in order to get it working. It'd be great to hear how you go!

**Reply by svarga91** [reply]
I did a little more research and found that only http has this login problem. I wanted to use https anyway.

**Reply by toddca** [reply]
I ran into the same issue. My problem was I set the SEQ_API_CANONICALURI to a HTTPS URI vs a HTTP Url and I had configured docker port mapping 80:80. Once I changed my SEQ_API_CANONICALURI to a HTTP URI things worked.

**Comment by Reyhn3** [comment]
Late to the party, but I had the same problem as the OP.

This was caused by the initial admin password was **less than 8 characters long**.

Thus the solution was to use a longer password.
I didn't find any documentation that mentions any password requirements.

**Reply by nblumhardt** [reply]
Interesting! There shouldn't be a lower limit on this, but we'll investigate - might be something unexpected in there 👍

**Reply by Reyhn3** [reply]
Sorry if I'm leading you on a wild goose chase now, because I am unable to reproduce it today.

Yesterday, when I was following the [Getting Started](https://hub.docker.com/r/datalust/seq) instructions, I set the admin password to `admin`. When I then tried to log in with username `admin` and password `admin` nothing at all happened on the login page. I don't recall where I saw the error (it might have been in the docker logs or in the browser console), but I am **certain** I saw in plain text that the password must be a minimum of "8 characters". That's what prompted me to redo the procedure with the very imaginative password `adminadmin` instead, which worked.

Today I can't find it. ): But I remember yesterday thinking that the error message should have been displayed somewhere, or the password requirement documented, as the feedback was very well hidden.

**Reply by nblumhardt** [reply]
Thanks for the follow-up. A few times, locally while spinning up the UI, I've noticed what might be a race condition in the login screen, where login completes but the "Events" screen route doesn't load - so you're left with the full logged-in navigation bar, but the login form still displayed.

I don't suppose that could be the cause of the issues in this thread? (Given enough login attempts by enough people we'll probably repro just about any lurking bug...) I'll take a closer look at this possibility.
